src/containerapp-preview/azext_containerapp_preview/_validators.py

```diff
@@ -0,0 +1,54 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from ._constants import (MANAGED_ENVIRONMENT_TYPE,
+                         CONNECTED_ENVIRONMENT_TYPE,
+                         MANAGED_ENVIRONMENT_RESOURCE_TYPE,
+                         CONNECTED_ENVIRONMENT_RESOURCE_TYPE,
+                         CONTAINERAPP_NAMESPACE)
+
+
+def validate_env_name_or_id(cmd, namespace):
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    from msrestazure.tools import is_valid_resource_id, resource_id, parse_resource_id
+
+    if not namespace.managed_env or not namespace.resource_group_name:
+        return
+
+    # Set environment type
+    environment_type = None
+
+    if namespace.__dict__.get("environment_type"):
+        environment_type = namespace.environment_type
+
+    if namespace.managed_env:
+        if is_valid_resource_id(namespace.managed_env):
+            env_dict = parse_resource_id(namespace.managed_env)
+            resource_type = env_dict.get("resource_type")
+            if resource_type:
+                if CONNECTED_ENVIRONMENT_RESOURCE_TYPE.lower() == resource_type.lower():
+                    environment_type = CONNECTED_ENVIRONMENT_TYPE
+                if MANAGED_ENVIRONMENT_RESOURCE_TYPE.lower() == resource_type.lower():
+                    environment_type = MANAGED_ENVIRONMENT_TYPE
+    if namespace.__dict__.get("custom_location") or namespace.__dict__.get("connected_cluster_id"):
```

Comment:
add comment for custom location and connected cluster id. comment unused code and leave comment for what to do